pkg/kubelet/eviction/types.go

```diff
@@ -67,3 +77,21 @@ type Threshold struct {
 // status - the desired status to associate with the pod (i.e. why its killed)
 // gracePeriodOverride - the grace period override to use instead of what is on the pod spec
 type KillPodFunc func(pod *api.Pod, status api.PodStatus, gracePeriodOverride *int64) error
+
+// ActivePodsFunc returns pods bound to the kubelet that are active (i.e. non-terminal state)
+type ActivePodsFunc func() []*api.Pod
+
+// statsFunc returns the usage stats if known for an input pod.
+type statsFunc func(pod *api.Pod) (statsapi.PodStats, bool)
+
+// rankFunc sorts the pods in eviction order
+type rankFunc func(pods []*api.Pod, stats statsFunc)
+
+// signalObservations maps a signal to an observed quantity
+type signalObservations map[Signal]resource.Quantity
+
+// thresholdsObservedAt maps a threshold to a time that it was observed
+type thresholdsObservedAt map[Threshold]time.Time
```

Comment:
I think this is broken, because you're assuming that resource.Quantity contributes to the uniqueness of the the threshold.  It does not, because it's a pointer to an inf.Dec, which is only compared on pointer equality, not actual equality.

If you are depending on uniqueness of Quantity to signal value, you'll need to change your map keys.